Libraries/src/Amazon.Lambda.Annotations.SourceGenerator/Writers/CloudFormationWriter.cs

```diff
@@ -0,0 +1,323 @@
+            _templateWriter.SetToken($"{lambdaFunctionPath}.Metadata.Tool", "Amazon.Lambda.Annotations");
+
+            _templateWriter.SetToken($"{propertiesPath}.Runtime", "dotnet6");
+            _templateWriter.SetToken($"{propertiesPath}.CodeUri", "");
+            _templateWriter.SetToken($"{propertiesPath}.MemorySize", 256);
+            _templateWriter.SetToken($"{propertiesPath}.Timeout", 30);
+            _templateWriter.SetToken($"{propertiesPath}.Policies", new List<string>{"AWSLambdaBasicExecutionRole"}, TokenType.List);
+        }
+
+        /// <summary>
+        /// Creates a new serverless template with no resources.
+        /// </summary>
+        private void CreateNewTemplate()
+        {
+            _templateWriter.SetToken("AWSTemplateFormatVersion", "2010-09-09");
+            _templateWriter.SetToken("Transform", "AWS::Serverless-2016-10-31");
+        }
+
+        /// <summary>
+        /// Removes all Lambda functions that exist in the serverless template but were not encountered during the current source generation pass.
+        /// Any resource that is removed must be of type 'AWS::Serverless::Function' and must have 'Resources.FUNCTION_NAME.Metadata.Tool' == 'Amazon.Lambda.Annotations'.
+        /// </summary>
+        private void RemoveOrphanedLambdaFunctions(HashSet<string> processedLambdaFunctions)
+        {
+            if (!_templateWriter.Exists("Resources"))
+            {
+                return;
+            }
+
+            var toRemove = new List<string>();
+            foreach (var resourceName in _templateWriter.GetToken<Dictionary<string, object>>("Resources").Keys)
+            {
+                var resourcePath = $"Resources.{resourceName}";
+                var type = _templateWriter.GetToken<string>($"{resourcePath}.Type", string.Empty);
+                var creationTool = _templateWriter.GetToken<string>($"{resourcePath}.Metadata.Tool", string.Empty);
+
+                if (string.Equals(type, "AWS::Serverless::Function", StringComparison.Ordinal)
+                    && string.Equals(creationTool, "Amazon.Lambda.Annotations", StringComparison.Ordinal)
+                    && !processedLambdaFunctions.Contains(resourceName))
+                {
+                    toRemove.Add(resourceName);
+                }
+            }
+
+            foreach (var resourceName in toRemove)
+            {
+                _templateWriter.RemoveToken($"Resources.{resourceName}");
+            }
+        }
+
+        /// <summary>
+        /// If the string does not start with '@', return it as is.
+        /// If a string value starts with '@' that means it is specified by another resource in the serverless template and a reference node is created and returned.
+        /// </summary>
+        private object GetValueOrRef(string value)
+        {
+            if (!value.StartsWith("@"))
+                return value;
+
+            switch (_templateFormat)
```

Comment:
This is trending towards academia, but if `CloudFormationWriter` is starting to have to make decisions between JSON and YAML we may be leaking an abstraction.

Can this be moved to `ITemplateWriter`, and then each implementation implements its own case from here?

Reply:
Agreed this should be pushed into the `ITemplateWriter`. CloudFormationWriter should not have to be working with the JSON or YAML types directly.

Reply:
Moved this method to `ITemplateWriter`